router/http_test.go

```diff
@@ -339,6 +339,34 @@ func (s *S) TestHTTPHeadersFromClient(c *C) {
 	c.Assert(res.StatusCode, Equals, 200)
 }
 
+func (s *S) TestConnectionCloseHeaderFromClient(c *C) {
+	h := http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		// Connection: close header should be stripped by the reverse proxy so it
+		// always does keep-alive with backends.
+		c.Assert(req.Close, Equals, false)
+		c.Assert(req.Header.Get("Proxy-Authenticate"), Equals, "")
+		c.Assert(req.Header.Get("Proxy-Authorization"), Equals, "")
```

Comment:
yeah, that's a great question. I didn't do any research on the `Proxy-` headers, just left them in the list because they were in `httputil.ReverseProxy`. It certainly sounds like those shouldn't be removed by that proxy either.